synapse/lib/task.py

```diff
@@ -96,6 +96,9 @@ def pack(self):
 
         return pask
 
+    def cancel(self):
+        return self.task.cancel()
```

Comment:
This API is  appears unused - what is it here for?

Reply:
This is a fix for the fact that the fini() routine in the agenda subsystem calls cancel() on the synapse Task objects, which have no cancel() method.

Reply:
( adding a test to demonstrate )